physical/mssql/mssql.go

```diff
@@ -59,43 +51,50 @@ func NewMSSQLBackend(conf map[string]string, logger log.Logger) (physical.Backen
 		maxParInt = physical.DefaultParallelOperations
 	}
 
-	database, ok := conf["database"]
-	if !ok {
-		database = "Vault"
-	}
-
-	table, ok := conf["table"]
-	if !ok {
-		table = "Vault"
-	}
-
-	appname, ok := conf["appname"]
-	if !ok {
-		appname = "Vault"
+	// set defaults
+	defaults := map[string]string{
+		"database":           "Vault",
+		"table":              "Vault",
+		"appname":            "Vault",
+		"connection timeout": "30",
+		"schema":             "dbo",
+		"log level":          "0",
 	}
 
-	connectionTimeout, ok := conf["connectiontimeout"]
-	if !ok {
-		connectionTimeout = "30"
+	// backcomp maps commonly used hcl paramater names to valid sql server parameters for backwards compatibility
+	backcomp := map[string]string{
+		"connectiontimeout": "connection timeout",
+		"loglevel":          "log level",
+		"username":          "user id",
 	}
 
-	logLevel, ok := conf["loglevel"]
-	if !ok {
-		logLevel = "0"
-	}
-
-	schema, ok := conf["schema"]
-	if !ok || schema == "" {
-		schema = "dbo"
+	// upgrade parameter keys
+	var isSet bool
```

Comment:
Removing these and using a short declaration in the `for` loop would be preferable.